src/rgw/services/svc_sys_obj.cc

```diff
@@ -0,0 +1,206 @@
+{
+  return Obj(core_svc.get(), obj_ctx, obj);
+}
+
+int RGWS_SysObj::create_instance(const string& conf, RGWServiceInstanceRef *instance)
+{
+  instance->reset(new RGWSI_SysObj(this, cct));
+  return 0;
+}
+
+std::map<string, RGWServiceInstance::dependency> RGWSI_SysObj::get_deps()
+{
+  RGWServiceInstance::dependency dep1 = { .name = "rados",
+                                          .conf = "{}" };
+  RGWServiceInstance::dependency dep2 = { .name = "sysobj_core",
+                                          .conf = "{}" };
+  map<string, RGWServiceInstance::dependency> deps;
+  deps["rados_dep"] = dep1;
+  deps["sysobj_core_dep"] = dep2;
+  return deps;
+}
+
+int RGWSI_SysObj::load(const string& conf, std::map<std::string, RGWServiceInstanceRef>& dep_refs)
+{
+  rados_svc = static_pointer_cast<RGWSI_RADOS>(dep_refs["rados_dep"]);
+  assert(rados_svc);
+
+  core_svc = static_pointer_cast<RGWSI_SysObj_Core>(dep_refs["sysobj_core_dep"]);
+  assert(core_svc);
+
+  return 0;
+}
+
+void RGWSI_SysObj::Obj::invalidate_state()
+{
+  ctx.invalidate(obj);
+}
+
+int RGWSI_SysObj::Obj::ROp::stat()
+{
+  RGWSI_SysObj_Core *svc = source.core_svc;
+  rgw_raw_obj& obj = source.obj;
+
+  return svc->stat(source.get_ctx(), state, obj, attrs,
+                   lastmod, obj_size,
+                   objv_tracker);
+}
+
+int RGWSI_SysObj::Obj::ROp::read(int64_t ofs, int64_t end, bufferlist *bl)
+{
+  RGWSI_SysObj_Core *svc = source.core_svc;
+  rgw_raw_obj& obj = source.get_obj();
+
+  return svc->read(source.get_ctx(), state,
+                   objv_tracker,
+                   obj, bl, ofs, end,
+                   attrs,
+                   cache_info,
+                   refresh_version);
+}
+
+int RGWSI_SysObj::Obj::ROp::get_attr(const char *name, bufferlist *dest)
+{
+  RGWSI_SysObj_Core *svc = source.core_svc;
+  rgw_raw_obj& obj = source.get_obj();
+
+  return svc->get_attr(obj, name, dest);
+}
+
+int RGWSI_SysObj::Obj::WOp::remove()
+{
+  RGWSI_SysObj_Core *svc = source.core_svc;
+  rgw_raw_obj& obj = source.get_obj();
+
+  return svc->remove(source.get_ctx(),
+                     objv_tracker,
+                     obj);
+}
+
+int RGWSI_SysObj::Obj::WOp::write(bufferlist& bl)
+{
+  RGWSI_SysObj_Core *svc = source.core_svc;
+  rgw_raw_obj& obj = source.get_obj();
+
+  return svc->write(obj, pmtime, attrs, exclusive,
+                    bl, objv_tracker, mtime);
+}
+
+int RGWSI_SysObj::Obj::WOp::write_data(bufferlist& bl)
+{
+  RGWSI_SysObj_Core *svc = source.core_svc;
+  rgw_raw_obj& obj = source.get_obj();
+
+  return svc->write_data(obj, bl, exclusive, objv_tracker);
+}
+
+int RGWSI_SysObj::Obj::WOp::write_attrs()
+{
+  RGWSI_SysObj_Core *svc = source.core_svc;
+  rgw_raw_obj& obj = source.get_obj();
+
+  return svc->set_attrs(obj, attrs, nullptr, objv_tracker);
+}
+
+int RGWSI_SysObj::Pool::Op::list_prefixed_objs(const string& prefix, list<string> *result)
+{
+  bool is_truncated;
+
+  auto rados_pool = source.rados_svc->pool(source.pool);
```

Comment:
should probably have SysObj_Core just a pure interface, abd gave SysObj_RADOS as the RADOS implementation. The pool iteration stuff will probably belong there. There are still all the `RGWRados::pool_iterate* ` methods that need to be moved, so that would be the time to clean this.